From 847310f9bb91f472a939e5dcd731cd490d73914b Mon Sep 17 00:00:00 2001
From: Sylvain Ageneau <ageneau@gmail.com>
Date: Mon, 3 Dec 2012 17:19:53 -0300
Subject: [PATCH 4/4] Load dynamic libraries lazyly so that we can
 cross-compile cffi libraries without the cross-compiler
 having to resolve all the symbols of the wrapped
 library

---
 src/c/ffi/libraries.d |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/c/ffi/libraries.d b/src/c/ffi/libraries.d
index 82cb931..33a00c2 100644
--- a/src/c/ffi/libraries.d
+++ b/src/c/ffi/libraries.d
@@ -182,7 +182,7 @@ dlopen_wrapper(cl_object block)
 	cl_object filename = block->cblock.name;
         char *filename_string = (char*)filename->base_string.self;
 #ifdef HAVE_DLFCN_H
-	block->cblock.handle = dlopen(filename_string, RTLD_NOW|RTLD_GLOBAL);
+	block->cblock.handle = dlopen(filename_string, RTLD_LAZY|RTLD_GLOBAL);
 #endif
 #ifdef HAVE_MACH_O_DYLD_H
 	{
-- 
1.7.10.4

